### PR TITLE
Release workflow

### PR DESCRIPTION
**Pull request by pkoryzna**

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -90,4 +90,12 @@ Connect to LDAP server (e.g. with [Apache Directory Studio](﻿http://directory.
 or browse [example.ldif](docker/bootstrap/example.ldif) LDIF file for available usernames.
   
 Before changing Cassandra authenticator implementation you can execute [roles.cql](docker/roles.cql) 
-file to configure the available roles (instead of creating them manually).  
+file to configure the available roles (instead of creating them manually).  ## Release process
```
Comment (alex-vv) [inline README.MD]: please add some newlines before the heading

**Review by alex-vv — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -1 +1,12 @@
 language: java
+install: mvn install -DskipTests=true -DfinalName=cassandra-shiro-$TRAVIS_TAG
+script: mvn verify -DfinalName=cassandra-shiro-$TRAVIS_TAG
+deploy:
+  skip_cleanup: true
+  provider: releases
+  api_key:
+    secure: cvh03OXEwdmvmPpohRbM5CChht6sXRfQzBGWDT7kiL/KxEWAY78sphoJQpJwqxKGYOib2CqMQcT6ffoXhl/75VOnXEFFo6oU1eubkMLcoA8FAQuMYyT+TXLC1qkM6yzT58pDBobqTqulA5t0XmPQlyoDIFmvRvI00nrUkv72T3hsevqazQ9PFzVuUIG0LqFbcnOKkC6yXNeY4VkOYd5V1u0HSoFIXnf/ga0E5F7D+lwE0pb5GQ/xX1LdifQukTMUFOgn02zYhbrbrjvATwOuRmDhYme1KkX3pHeuKXKXKfpKJfDUiqFqOHXDFzWZyhfASaqKkNyy/C8jwfGBklaEvA6mpIPvnxu6a7W8ZtZO8Wl0gs+cQST1aT819MVFdX+NgVVTT2QiykKEbKsDehU3Gw5sXOQlgKqAYLmuJhdofXrSQv+euTw0VfIuiPFC4ENr4yZoMuuGlY2sQwgfKJVwIqqTs7DltAyjuaJ9EEYyojMYOewmL+N/IeWfaRVCxMaTp3ofsfKOx7hN2+fEM/KuEzugHGG3PPJWq1BplutB5irEO2Mv5O+AMVPVC9J3bw8scbZOgzOgM2ZAYgxF4eTm4i2kEpjn+ZXrn6EJYZlecwx6aXJ4NFuipGHbDK66wWurAioydWVK5tNadBdABJvd6h5528NMKQnT5wkUFcdTsTU=
```
Comment (juwalter) [inline .travis.yml:8]: hm, what is this for - it is in the repo for everybody to see, right?

**Comment by juwalter** [comment]
can you please squash the commits before merging and also add a description to this PR ie what it will accomplish and possibly some reasoning etc?

**Comment by juwalter** [comment]
crap- closed by accident

File: .travis.yml (hunk 2 above)
Comment (pkoryzna) [reply .travis.yml:8]: it's  encrypted api key for github releases https://docs.travis-ci.com/user/encryption-keys/

**Review by alex-vv — APPROVED** [review]

File: README.MD (hunk 1 above)
Comment (pkoryzna) [reply README.MD]: missed that - will fix

**Comment by pkoryzna** [comment]
@juwalter this is adding a config for releases prepared by following the steps in the README.MD

re: squash - we can squash commits directly in github

screen shot 2018-03-01 at 12 15 42

**Comment by alex-vv** [comment]
@pkoryzna I wonder if you squash commits now would not `1.0.0` release tag be lost?

**Comment by pkoryzna** [comment]
@alex-vv Good point. I think it'd stay but detached from any branch. So I'd say we merge it as normal.

**Comment by juwalter** [comment]
I think it is better to do the squashing manually- in a "semantic" way- not like automatically all into one

also, it is nicer for doing a review if the commits are grouped semantically and named/messaged accordingly - what do you think?

**Comment by pkoryzna** [comment]
@juwalter can you explain what you mean by "semantically named commits" in this case?

**Comment by juwalter** [comment]
@pkoryzna 

by "semantically named commits" I mean

1. squash those together that clearly belong together: "belong together" = based on semantics/meaning - e.g. "add final name property to pom" is sitting there quite isolated, I am sure it belongs somewhat to "updating pom" etc.
2. commit message - should more describe the meaning of the commit, e.g. "update settings for travis continuous build system to push generated artifacts to a releases URI" 

makes sense?

**Comment by pkoryzna** [comment]
so more general commit messages?

**Comment by juwalter** [comment]
yes, sounds good!

**Pull request merged**